lib/events.js

```diff
@@ -1169,7 +1169,7 @@ function on(emitter, event, options = kEmptyObject) {
   }
   if (signal) {
     kResistStopPropagation ??= require('internal/event_target').kResistStopPropagation;
-    eventTargetAgnosticAddListener(
+    addEventListener(
```

Comment:
This would work, but truth be told, it should probably be replaced with `addAbortListener`, which will return a disposable object which we can dispose of in the `closeHandler` function.
```javascript
addAbortListener(signal, abortListener);
```

Reply:
The one line change was too good to be true! Happy to make that change, thanks for the feedback.